### PR TITLE
update CI Purescript version

### PR DESCRIPTION
**Pull request by peterbecich**
**Description of the change**
Updated CI from Purescript 0.14.0-rc5 to 0.14.0

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a link to this PR and your username
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation in the README and/or documentation directory
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by thomashoneyman** [comment]
We'll need to update the `packages.dhall` as well, for a 0.14-compatible package set. For example:

```dhall
let upstream =
      https://github.com/purescript/package-sets/releases/download/psc-0.14.0/packages.dhall sha256:710b53c085a18aa1263474659daa0ae15b7a4f453158c4f60ab448a6b3ed494e

in  upstream
```

**Comment by thomashoneyman** [comment]
This _may_ need other updates for PureScript 0.14 compatibility -- looks like this library was missed in the updates.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
       - name: Set up PureScript toolchain
         uses: purescript-contrib/setup-purescript@main
         with:
-          purescript: "0.14.0-rc5"
+          purescript: "0.14.0"
```
Comment (thomashoneyman) [inline .github/workflows/ci.yml:15]: As with your other PR, these two lines can be removed as setup-purescript will pull in the latest compiler automatically.

File: packages.dhall [hunk 2]
```diff
@@ -1,4 +1,4 @@
 let upstream =
-      https://raw.githubusercontent.com/purescript/package-sets/prepare-0.14/src/packages.dhall
+      https://github.com/purescript/package-sets/releases/download/psc-0.14.0-20210304/packages.dhall sha256:c88151fe7c05f05290224c9c1ae4a22905060424fb01071b691d3fe2e5bad4ca
```
Comment (peterbecich) [inline packages.dhall:2]: Does it matter if the set is `psc-0.14.0-20210304` or `psc-0.14.0`?

https://github.com/purescript-contrib/purescript-github-actions-toolkit/pull/14#issuecomment-792227946

Comment (thomashoneyman) [reply packages.dhall:2]: It doesn't matter -- I have a _slight_ preference for 0.14.0, but I'll accept either.

Comment (thomashoneyman) [reply packages.dhall:2]: Also, my comment on the changelog applies to each of these pull requests. Sorry I didn't see it sooner!

**Review by thomashoneyman — APPROVED** [review]

**Pull request merged**